app/code/core/Mage/Adminhtml/Block/Widget/Grid.php

```diff
@@ -1289,12 +1289,18 @@ public function getParam($paramName, $default=null)
     {
         $session = Mage::getSingleton('adminhtml/session');
         $sessionParamName = $this->getId().$paramName;
-        if ($this->getRequest()->has($paramName)) {
-            $param = $this->getRequest()->getParam($paramName);
-            if ($this->_saveParametersInSession) {
-                $session->setData($sessionParamName, $param);
+        $param = $this->getRequest()->getParam($paramName);
+
+        if ($param !== null) {
+            if (trim($param) !== '') {
```

Comment:
Can param also be an array in this context?

Reply:
no it can't be an array - multiple filter will serialize to a string. sorry for my lazy response :)